packages/apps/shopify-app-remix/src/server/authenticate/admin/billing/create-usage-record.ts

```diff
@@ -0,0 +1,40 @@
+import {HttpResponseError, Session} from '@shopify/shopify-api';
+
+import type {BasicParams} from '../../../types';
+import {redirectToAuthPage} from '../helpers';
+import {invalidateAccessToken} from '../../helpers';
+
+import type {CreateUsageRecordOptions} from './types';
+
+export function createUsageRecordFactory(
+  params: BasicParams,
+  request: Request,
+  session: Session,
+) {
+  return async function createUsageRecord(options: CreateUsageRecordOptions) {
+    const {api, logger} = params;
+
+    logger.debug('Create usage record', {shop: session.shop, ...options});
+
+    try {
+      return await api.billing.createUsageRecord({
+        session,
+        description: options.description,
+        price: options.price,
+        isTest: options.isTest,
+        idempotencyKey: options.idempotencyKey,
+        subscriptionLineItemId: options.subscriptionLineItemId,
+      });
```

Comment:
Any particular reason not to pass the options through?

```suggestion
      return await api.billing.createUsageRecord({
        ...options,
        session,
      });
```